### PR TITLE
SMHE-729: Fix the schedule test alarm by correcting date time structure

### PR DESCRIPTION
**Pull request by abolder**
Signed-off-by: Andre Bolder <andre.bolder@alliander.com>
Changed the data structure from datetime to seperate time and date data structure.
Added a JUnit test for this command executer.

### CODE REVIEW

**Comment by abolder** [comment]
✔︎ https://jenkins.fdp.osgp.cloud/view/Nightly%20Build/job/gxf-nightly-build/524/

**Review by natasja-n — APPROVED** [review]

File: osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/alarm/TestAlarmSchedulerCommandExecutorTest.java [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+
+    this.connectionStub.setDefaultReturnValue(DataObject.newArrayData(Collections.emptyList()));
+  }
+
+  @ParameterizedTest
+  @CsvSource({
+    "EMPTY,LAST_GASP,No scheduled date-time set", //
+    "EMPTY,PARTIAL_POWER_OUTAGE,No scheduled date-time set",
+    "2020-10-07 15:33:00,PARTIAL_POWER_OUTAGE,Incorrect scheduled date time value set. It should not be a past date",
+    "2020-10-07 15:33:00,LAST_GASP,Incorrect scheduled date time value set. It should not be a past date"
```
Comment (smvdheijden) [inline osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/alarm/TestAlarmSchedulerCommandExecutorTest.java]: Remove `//` at end of first line.
Consider adding a test for an empty alarmtype as well.

Comment (abolder) [reply osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/alarm/TestAlarmSchedulerCommandExecutorTest.java]: Added the suggested test cases and removed the // formatting.

File: osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/alarm/TestAlarmSchedulerCommandExecutorTest.java [hunk 2]
```diff
@@ -83,15 +83,17 @@ void setup() {
...
   @ParameterizedTest
   @CsvSource({
-    "EMPTY,LAST_GASP,No scheduled date-time set", //
+    "EMPTY,LAST_GASP,No scheduled date-time set",
     "EMPTY,PARTIAL_POWER_OUTAGE,No scheduled date-time set",
     "2020-10-07 15:33:00,PARTIAL_POWER_OUTAGE,Incorrect scheduled date time value set. It should not be a past date",
-    "2020-10-07 15:33:00,LAST_GASP,Incorrect scheduled date time value set. It should not be a past date"
+    "2020-10-07 15:33:00,LAST_GASP,Incorrect scheduled date time value set. It should not be a past date",
+    "2060-10-07 15:33:00,EMPTY,No alarmtype set",
+    "2060-10-07 15:33:00,EMPTY,No alarmtype set"
```
Comment (smvdheijden) [inline osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/alarm/TestAlarmSchedulerCommandExecutorTest.java]: Twice the same test?

Comment (abolder) [reply osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/alarm/TestAlarmSchedulerCommandExecutorTest.java]: No of course not. I will remove one.

Comment (smvdheijden) [reply osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/alarm/TestAlarmSchedulerCommandExecutorTest.java]: :+1:
